src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,149 @@
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.login = null");
+    }
+
+    @Test
+    void register_ageIsNull_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin(LOGIN_FIRST);
+            defaultUser.setAge(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.age = null");
+    }
+
+    @Test
+    void register_passwordIsNull_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setPassword(null);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException expected for user.password = null");
+    }
+
+    @Test
+    void register_loginAlreadyInStorage_notOk() {
+        User storageUser = new User();
+        storageUser.setLogin(LOGIN_SECOND);
+        storageDao.add(storageUser);
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin(LOGIN_SECOND);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, method allows user duplicates");
+    }
+
+    @Test
+    void register_loginIsEmpty_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin("");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException,"
+                + "when login field empty an exception should be thrown");
+    }
+
+    @Test
+    void register_default_Ok() {
+        registrationService.register(defaultUser);
+        assertEquals(storageDao.get(defaultUser.getLogin()).getLogin(), defaultUser.getLogin());
+        assertEquals(storageDao.get(defaultUser.getLogin()).getPassword(),
+                defaultUser.getPassword());
+        assertEquals(storageDao.get(defaultUser.getLogin()).getAge(), defaultUser.getAge());
+    }
+
+    @Test
+    void register_passwordLengthLessThan6_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin(LOGIN_THIRD);
+            defaultUser.setPassword("");
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException,"
+                + "if password length less than 6 an exception should be thrown");
+    }
+
+    @Test
+    void register_AgeLessThan18_notOk() {
+        assertThrows(InvalidRegistrationDataException.class, () -> {
+            defaultUser.setLogin(LOGIN_FOURTH);
+            defaultUser.setAge(AGE_FIRST);
+            registrationService.register(defaultUser);
+        }, "InvalidRegistrationDataException, if age < 18 method you should thrown an exception");
+    }
+
+    @Test
+    void register_AgeEquals18_Ok() {
```

Comment:
next word after the underscore `_` should start from the lowercase letter. Fix it in all places 
```suggestion
    void register_ageEquals18_Ok() {
```